components/core/src/clp_s/SchemaTree.hpp

```diff
@@ -21,21 +22,37 @@ enum class NodeType : uint8_t {
     NullValue,
     DateString,
     StructuredArray,
+    Internal,
```

Comment:
Maybe we can add a description somewhere to explain what an internal node is. And do you think metadata node sounds more intuitive because internal nodes also refer to nodes that have children (the opposite of leaf node)?

Reply:
That's fair. We can rename it to metadata node. I guess we could generally document usage of this enum in a comment and mention what the Metadata node type is for there.